### PR TITLE
[Hair] - Code cleaning, refactor, comments and prep for PR

### PR DESCRIPTION
**Pull request by Adi-Amazon**

### CODE REVIEW

File: Gems/AtomTressFX/Code/Rendering/HairFeatureProcessor.h [hunk 1]
```diff
@@ -41,8 +41,15 @@ namespace AZ
     {
         namespace Hair
         {
+            //! The following few lined dictates the overall memory consumption reserved for the
+            //!  PPLL fragments.  The memory consumption using this technique is quite large (can
+            //!  grow far above 1GB in GPU/CPU data and in extreme cases of zoom up with dense hair
+            //!  might still not be enough.  For this reason it is recommended to utilize the
+            //!  approximated lighting scheme originally suggested by Eidos Montreal and do OIT using
```
Comment (hershey5045) [inline Gems/AtomTressFX/Code/Rendering/HairFeatureProcessor.h:48]: Are there any guidelines to mentioning other legal entities in comments? It's probably not an issue but is worth double-checking.

File: Gems/AtomTressFX/External/Code/src/TressFX/TressFXSettings.h [hunk 2]
```diff
@@ -31,7 +31,6 @@
 /*
 #pragma once
 */
-//#include <TressFX/AMD_Types.h>
```
Comment (hershey5045) [inline Gems/AtomTressFX/External/Code/src/TressFX/TressFXSettings.h:34]: If we modify any files here, we might have to add license headers. Should we move these outside of external instead and into our code?

File: Gems/AtomTressFX/Assets/Shaders/HairSimulationCompute.azsl [hunk 3]
```diff
@@ -62,25 +62,22 @@ void IntegrationAndGlobalShapeConstraints(
 
     // Integrate
     float dampingCoeff = GetDamping(strandType);
-//    float4 oldPos = float4(CM_TO_METERS,CM_TO_METERS,CM_TO_METERS,1.0) * g_HairVertexPositionsPrev[globalVertexIndex];
```
Comment (hershey5045) [inline Gems/AtomTressFX/Assets/Shaders/HairSimulationCompute.azsl:65]: I missed adding headers in this file it seems, I can do a second pass! If you do have time to add, we want the modified header for this
```
/*
 * Modifications Copyright (c) Contributors to the Open 3D Engine Project. 
 * For complete copyright and license terms please see the LICENSE at the root of this distribution.
 * 
 * SPDX-License-Identifier: (Apache-2.0 OR MIT) AND MIT
 *
 */
```

File: Gems/AtomTressFX/Code/Rendering/HairFeatureProcessor.h (hunk 1 above)
Comment (Adi-Amazon) [reply Gems/AtomTressFX/Code/Rendering/HairFeatureProcessor.h:48]: Good question but not as far as I'm aware as long as it's not offensive or code copy without authorization. In remarks it should be ok to recommend a technique and / or have a reference link (like we do in Atom shaders in many locations).

File: Gems/AtomTressFX/Assets/Shaders/HairSimulationCompute.azsl (hunk 3 above)
Comment (Adi-Amazon) [reply Gems/AtomTressFX/Assets/Shaders/HairSimulationCompute.azsl:65]: Added, good catch.

File: Gems/AtomTressFX/External/Code/src/TressFX/TressFXSettings.h (hunk 2 above)
Comment (Adi-Amazon) [reply Gems/AtomTressFX/External/Code/src/TressFX/TressFXSettings.h:34]: Not required as it is under External and according to their license.
Unless went through major changes we keep the AMD original files under External and by that simplify the legal process of using their open source license that can be found under this directory.

**Pull request merged**